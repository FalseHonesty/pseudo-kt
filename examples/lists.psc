myList = [1, 2, 3, 4, 5] // List literal
print(myList)

myList.append(10) // Adds 10 to the end of the list
print(myList)
print(myList.length)

myList.pop(0) // Removes the first item in the list
print(myList)
print(myList.length)

myList.insert(1, 600) // Inserts the element '600' into the list so that it has the index 1
print(myList)
print(myList.length)

myList.remove(3) // Looks for the element '3' in the list and removes the first occurrence of it
print(myList)
print(myList.length)

print(myList.contains(600)) // Checks whether an element is in a list
print(myList.contains("hi"))

print(myList[1]) // Gets the item in the list with index 1

myList[2] = -50 // Overwrites the item in the list with index 2
print(myList)

myList[3] += 100 // Modifies the item in the list with index 3
print(myList)

// Nested lists:
nestedList = [[1, 2, 3], [4, 5, 6], [7, 8, 9]] // A 2D list
print(nestedList)

print(nestedList[0, 2]) // Get the item with the "coordinates" (0, 2)

nestedList[0, 2] += 1000 // Modify the item
print(nestedList)

print(nestedList[0][2]) // Slightly more explicit syntax, if you prefer. The two are equivalent