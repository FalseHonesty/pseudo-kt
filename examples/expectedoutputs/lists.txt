[1, 2, 3, 4, 5]
[1, 2, 3, 4, 5, 10]
6
[2, 3, 4, 5, 10]
5
[2, 600, 3, 4, 5, 10]
6
[2, 600, 4, 5, 10]
5
true
false
600
[2, 600, -50, 5, 10]
[2, 600, -50, 105, 10]
[[1, 2, 3], [4, 5, 6], [7, 8, 9]]
3
[[1, 2, 1003], [4, 5, 6], [7, 8, 9]]
1003
